python_modules/dagster/dagster/_daemon/auto_run_reexecution/auto_run_reexecution.py

```diff
@@ -25,66 +19,10 @@ def filter_runs_to_should_retry(
     runs: Sequence[DagsterRun], instance: DagsterInstance, default_max_retries: int
 ) -> Iterator[Tuple[DagsterRun, int]]:
     """Return only runs that should retry along with their retry number (1st retry, 2nd, etc.)."""
-
-    def get_retry_number(run: DagsterRun) -> Optional[int]:
-        if run.status != DagsterRunStatus.FAILURE:
-            return None
-
-        raw_max_retries_tag = run.tags.get(MAX_RETRIES_TAG)
-        if raw_max_retries_tag is None:
-            max_retries = default_max_retries
-        else:
-            try:
-                max_retries = int(raw_max_retries_tag)
-            except ValueError:
-                instance.report_engine_event(
-                    f"Error parsing int from tag {MAX_RETRIES_TAG}, won't retry the run.", run
-                )
-                return None
-
-        if max_retries == 0:
-            return None
-
-        # TODO: group these to reduce db calls
-        run_group = instance.get_run_group(run.run_id)
-
-        if run_group:
-            _, run_group_iter = run_group
-            run_group_list = list(run_group_iter)
-
-            # Has the parent run already been retried the maximum number of times? (Group includes the parent)
-            if len(run_group_list) >= max_retries + 1:
-                return None
-
-            # Does this run already have a child run?
-            if any([run.run_id == run_.parent_run_id for run_ in run_group_list]):
-                return None
-            return len(run_group_list)
-        else:
-            return 1
-
-    default_retry_on_asset_or_op_failure: bool = instance.get_settings("run_retries").get(
-        "retry_on_asset_or_op_failure", True
-    )
-
     for run in runs:
-        retry_number = get_retry_number(run)
-        retry_on_asset_or_op_failure = get_boolean_tag_value(
-            run.tags.get(RETRY_ON_ASSET_OR_OP_FAILURE_TAG),
-            default_value=default_retry_on_asset_or_op_failure,
-        )
-        if retry_number is not None:
-            if (
-                run.tags.get(RUN_FAILURE_REASON_TAG) == RunFailureReason.STEP_FAILURE.value
-                and not retry_on_asset_or_op_failure
-            ):
-                instance.report_engine_event(
-                    "Not retrying run since it failed due to an asset or op failure and run retries "
-                    "are configured with retry_on_asset_or_op_failure set to false.",
-                    run,
-                )
-            else:
-                yield (run, retry_number)
+        if get_boolean_tag_value(run.tags.get(WILL_RETRY_TAG), default_value=False):
```

Comment:
one thing to watch out for is that we probably do not want to copy over this tag in the set of tags that is set when a run is retried, or it will retry indefinitely - so we should probably add it here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/storage/tags.py#L105